src/Services/EnturSales.php

```diff
@@ -54,66 +42,61 @@ public static function dateTimeFormatter($date)
-        $chunkDate = Carbon::parse($chunkId)->format("Ymd");
-        $chunkDatePrevious = Carbon::parse($chunkId)->subDay()->format("Ymd");
-        $chunkIdPrevious = Carbon::parse($chunkId)->subDay()->format("Y-m-d");
-
-        // CLEOS won't start generating reports until first Working Day of the month.
-        // Because of this we need to check for corret file when fetching
-        // files for the first few days of any given month. We use ID of report (reportID)
-        // to fetch next, and use date given in filename to see if the corret report is given
-        while ($status == 200 && is_null($archive)) {
-            $reportID = intval($response->header("x-entur-report-id"));
-            $contentDisposition = $response->header("content-disposition");
-            $fileName = preg_split('/[ =]/', $contentDisposition, -1, PREG_SPLIT_NO_EMPTY)[2];
-
-            $this->debug("FILE: %s", $fileName);
-
-            $matches = null;
-            // match for date on format YYYY-MM-DD, YYYY/MM/DD, YYYY.MM.DD or YYYYMMDD
-            // matches years in range (inclusive) 2000-3999
-            preg_match(
-                '/[^\d](?<date>((2|3)\d{3})([-\/\.])*(0[1-9]|1[0,1,2])([-\/\.])*(0[1-9]|[12][0-9]|3[01]))/',
-                $fileName,
-                $matches
-            );
-
-            if (!isset($matches['date'])) {
-                return null;
-            }
-
-            // Fecth date match from preg_match, and make it same format as $chunkDate* variables
-            $fileDate = preg_replace('/[-\/\.]/', '', $matches['date']);
-
-            // Checking for either chunkId - 1day ($chunkDate) or chunkId (today)
-            // since there seems to be a tiny discrepency in when files are made
-            // available for download.
-            if (strcmp($chunkDate, $fileDate) == 0 || strcmp($chunkDatePrevious, $fileDate) == 0) {
-                $archive = new ChunkArchive(SinkEnturSales::$id, $chunkId);
-                $archive->addFromString($fileName, $response->body());
-                $archive->save();
-
-                $this->debug("Saving file %s, %s", $fileDate, $chunkDate);
-            } else {
-                $response = Http::withHeaders(['authorization' => 'Bearer ' . EnturCleosApi::getApiToken()])
-                    ->get($this->getCleosS1Url($chunkId, $reportID));
-                $status = $response->status();
-            }
+        $client = Http::withHeaders(['authorization' => 'Bearer ' . EnturCleosApi::getApiToken()]);
+        //$response = $client->get(EnturCleosApi::getNextDatasetInProduct(0, $chunkId));
+        //$nextDatasetResult = $client->get(EnturCleosApi::getNextDatasetInProductUrl(0, '2024-12-01'));
+        $nextDatasetResult = $client->get(EnturCleosApi::getNextDatasetInProductUrl(0, $chunkId));
+        $status = $nextDatasetResult->status();
+        if ($status != 200) {
+            $this->debug("STATUS nextDataset: %s", $status);
+            return null;
         }
```

Comment:
Should we throw here?

```suggestion
        $nextDatasetResult->throw();
```

Reply:
I have to look into throw (and/or maybe restructure the entire download function)